SteamKit2/SteamKit2/Util/NetHelpers.cs

```diff
@@ -9,36 +9,37 @@
 using System.Diagnostics.CodeAnalysis;
 using System.Net;
 using System.Net.Sockets;
+using System.Runtime.CompilerServices;
 using SteamKit2.Internal;
 
 namespace SteamKit2
 {
     static class NetHelpers
     {
-        public static IPAddress GetLocalIP(Socket activeSocket)
+        public static IPAddress GetLocalIP( Socket activeSocket )
         {
             var ipEndPoint = activeSocket.LocalEndPoint as IPEndPoint;
 
             if ( ipEndPoint == null || ipEndPoint.Address == IPAddress.Any )
                 throw new InvalidOperationException( "Socket not connected" );
-
+            
             return ipEndPoint.Address;
         }
 
         public static IPAddress GetIPAddress( uint ipAddr )
         {
-            byte[] addrBytes = BitConverter.GetBytes( ipAddr );
-            Array.Reverse( addrBytes );
-
+            Span<byte> addrBytes = stackalloc byte[ 4 ];
+            BitConverter.TryWriteBytes( addrBytes, IPAddress.NetworkToHostOrder( Unsafe.BitCast<uint, int>( ipAddr ) ) );
+            
             return new IPAddress( addrBytes );
```

Comment:
can't you construct from the int now without the byte array?

Reply:
has been changed to new IPAddress(long address)